ts/Extensions/Boost/BoostSeries.ts

```diff
@@ -235,6 +236,27 @@ function compose<T extends typeof Series>(
         ).forEach((method): void =>
             wrapSeriesFunctions(seriesProto, seriesTypes, method)
         );
+        wrap(
+            PointClass.prototype,
+            'firePointEvent',
+            function (
+                this: typeof PointClass.prototype,
+                proceed, type, e
+            ): boolean | undefined {
+                if (type === 'click' && this.series.boosted) {
+                    const point = e.point,
+                        { xAxis, yAxis } = point.series,
+                        distance = Math.sqrt(
+                            Math.pow(point.plotX + xAxis.pos - e.chartX, 2) +
+                            Math.pow(point.plotY + yAxis.pos - e.chartY, 2)
+                        );
```

Comment:
Some (if not all) points in this function already have `point.dist`. All points found using KD Tree are found through the distance calculation before entering here.